File: 13-roman-to-integer/13-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        int ans = 0;
        map<char,int> m;
        m['I'] = 1;
        m['V'] = 5;
        m['X'] = 10;
        m['L'] = 50;
        m['C'] = 100;
        m['D'] = 500;
        m['M'] = 1000;
        for(int i = 0; i < s.length(); i++) {
            if(s[i] == 'I' and i + 1 < s.length() and s[i+1] == 'V') {
                ans += 4;
                i++;
                continue;
            }
            
            if(s[i] == 'I' and i + 1 < s.length() and s[i+1] == 'X') {
                ans += 9;
                i++;
                continue;
            }
            
            if(s[i] == 'X' and i + 1 < s.length() and s[i+1] == 'L') {
                ans += 40;
                i++;
                continue;
            }
            
            if(s[i] == 'X' and i + 1 < s.length() and s[i+1] == 'C') {
                ans += 90;
                i++;
                continue;
            }
            
            if(s[i] == 'C' and i + 1 < s.length() and s[i+1] == 'D') {
                ans += 400;
                i++;
                continue;
            }
            
            if(s[i] == 'C' and i + 1 < s.length() and s[i+1] == 'M') {
                ans += 900;
                i++;
                continue;
            }
            ans += m[s[i]];
        }
        return ans;
    }
};

File: 13-roman-to-integer/13-roman-to-integer.py
class Solution:
    def romanToInt(self, roman_string: str) -> int:
        myDict = {}
        myDict["I"] = 1
        myDict["V"] = 5
        myDict["X"] = 10
        myDict["L"] = 50
        myDict["C"] = 100
        myDict["D"] = 500
        myDict["M"] = 1000
    
        integer = 0
        i = 0
    
        while i < len(roman_string):
            if roman_string[i] == "I":
                if i + 1 < len(roman_string) and (roman_string[i+1] == "X" or roman_string[i+1] == "V"):
                    integer += myDict[roman_string[i+1]] - myDict[roman_string[i]]
                    i += 1
                else:
                    integer += myDict[roman_string[i]]
            elif roman_string[i] == "X":
                if i + 1 < len(roman_string) and (roman_string[i+1] == "L" or roman_string[i+1] == "C"):
                    integer += myDict[roman_string[i+1]] - myDict[roman_string[i]]
                    i += 1
                else:
                    integer += myDict[roman_string[i]]
            elif roman_string[i] == "C":
                if i + 1 < len(roman_string) and (roman_string[i+1] == "D" or roman_string[i+1] == "M"):
                    integer += myDict[roman_string[i+1]] - myDict[roman_string[i]]
                    i += 1
                else:
                    integer += myDict[roman_string[i]]
            else:
                integer += myDict[roman_string[i]]
            i += 1
    

        return integer
        